test/integration/persistence_test.go

```diff
@@ -0,0 +1,62 @@
+// +build integration
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"path/filepath"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+
+	"k8s.io/minikube/test/integration/util"
+)
+
+func TestPersistence(t *testing.T) {
+	minikubeRunner := util.MinikubeRunner{BinaryPath: *binaryPath, T: t}
+	minikubeRunner.EnsureRunning()
+
+	kubectlRunner := util.NewKubectlRunner(t)
+	podName := "busybox"
+	podPath, _ := filepath.Abs("testdata/busybox.yaml")
+
+	podNamespace := kubectlRunner.CreateRandomNamespace()
+	defer kubectlRunner.DeleteNamespace(podNamespace)
+
+	// Create a pod and wait for it to be running.
+	if _, err := kubectlRunner.RunCommand([]string{"create", "-f", podPath, "--namespace=" + podNamespace}); err != nil {
+		t.Fatalf("Error creating test pod: %s", err)
+	}
+
+	p := &api.Pod{}
+	for p.Status.Phase != "Running" {
+		p = kubectlRunner.GetPod(podName, podNamespace)
+		time.Sleep(time.Second)
+	}
+
+	// Now restart minikube and make sure the pod is still there.
+	minikubeRunner.RunCommand("stop", true)
+	minikubeRunner.CheckStatus("Stopped")
+
+	minikubeRunner.RunCommand("start", true)
+	minikubeRunner.CheckStatus("Running")
+
+	kubectlRunner.GetPod(podName, podNamespace)
```

Comment:
Will it always be the case that the pod is up and running by the time this command is run?